src/utils/hooks/useNotificationsInfiniteScroll.tsx

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useContext, useEffect, useReducer, useRef } from 'react'
+
+import W3iContext from '@/contexts/W3iContext/context'
+import { notificationsReducer } from '@/reducers/notifications'
+
+const IntersectionObserverOptions = {
+  root: null,
+  rootMargin: '0px',
+  threshold: 1.0
+}
+
+export const useNotificationsInfiniteScroll = (topic?: string) => {
+  const intersectionObserverRef = useRef<HTMLDivElement>(null)
+  const { notifyClientProxy } = useContext(W3iContext)
+  const [state, dispatch] = useReducer(notificationsReducer, {})
+
+  const limit = 6
+
+  const nextPageInternal = useCallback(
+    async (lastMessageId?: string) => {
+      if (!(notifyClientProxy && topic)) {
+        return
+      }
+
+      const newNotifications = await notifyClientProxy.getNotificationHistory({
+        topic,
+        limit,
+        startingAfter: lastMessageId
+      })
+
+      dispatch({
+        type: 'FETCH_NOTIFICATIONS',
+        notifications: newNotifications.notifications,
+        uniqueIds: newNotifications.notifications.map(notification => notification.id),
```

Comment:
nit (non blocking): no need to pass `uniqueIds` since we can just calculate that in the reducer now

Reply:
Exactly!